crates/forge/src/lib.rs

```diff
@@ -72,6 +75,48 @@ struct TestsFromFile {
     relative_path: Utf8PathBuf,
 }
 
+#[must_use]
+pub fn collect_packages(
+    metadata: &Metadata,
+    manifest_path: &PathBuf,
+    package_name: &Option<String>,
+    workspace: bool,
+) -> Vec<PackageId> {
+    let workspace_members = &metadata.workspace.members;
+
+    let is_virtual_workspace = metadata
+        .packages
+        .iter()
+        .filter(|package| package.manifest_path == metadata.workspace.manifest_path)
+        .collect_vec()
+        .is_empty();
```

Comment:
Exactly, although I'll check again if there's more direct way to check it in the metadata.

Reply:
I asked Scarb team before and I don't think there is a direct method for that

Reply:
I think it could make sense for scarb metadata to expose this information. I'm assuming scarb is doing some checks if the workspace is virtual or not anyways.

cc @mkaput